src/record.c

```diff
@@ -292,6 +293,15 @@ int janus_recorder_add_extmap(janus_recorder *recorder, int id, const char *extm
 	return 0;
 }
 
+int janus_recorder_description(janus_recorder *recorder, const char *description) {
+	if (!description)
+		return -1;
```

Comment:
I'd add a

	janus_mutex_lock_nodebug(&recorder->mutex);

here (to unlock at the end before returning), and before replacing the string I'd also add a

	if(g_atomic_int_get(&recorder->header))

to return immediately `0` in case the header was already written (as in that case the description was already used and there's no point updating it).

Reply:
done